pkg/controller/livenessprobemapnodeprobe/probemapnodeprobe_utils.go

```diff
@@ -0,0 +1,223 @@
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileEnhancedLivenessProbeMapNodeProbe) retryOnConflictDelNodeProbeConfig(pod *v1.Pod) error {
+	nppClone := &appsv1alpha1.NodePodProbe{}
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: pod.Spec.NodeName}, nppClone); err != nil {
+			if errors.IsNotFound(err) {
+				klog.Warningf("No found getting updated npp %s from client", pod.Spec.NodeName)
+				return nil
+			}
+			klog.Errorf("Error getting updated npp %s from client", pod.Spec.NodeName)
+			return err
+		}
+
+		podNewNppCloneSpec := nppClone.Spec.DeepCopy()
+
+		newPodProbeTmp := []appsv1alpha1.PodProbe{}
+		for index := range podNewNppCloneSpec.PodProbes {
+			podContainerProbe := podNewNppCloneSpec.PodProbes[index]
+			if podContainerProbe.Name == pod.Name && podContainerProbe.Namespace == pod.Namespace &&
+				podContainerProbe.UID == fmt.Sprintf("%v", pod.UID) {
+				continue
+			}
+			newPodProbeTmp = append(newPodProbeTmp, podContainerProbe)
+		}
+		podNewNppCloneSpec.PodProbes = newPodProbeTmp
+
+		// delete node pod probe
+		if len(podNewNppCloneSpec.PodProbes) == 0 {
+			if err := r.Delete(context.TODO(), nppClone); err != nil {
+				klog.Errorf("Failed to delete node pod probe for pod: %s/%s, err: %v", pod.Namespace, pod.Name, err)
+				return err
+			}
+			return nil
+		}
+		if reflect.DeepEqual(podNewNppCloneSpec, nppClone.Spec) {
+			return nil
+		}
+		nppClone.Spec = *podNewNppCloneSpec
+		return r.Client.Update(context.TODO(), nppClone)
+	})
+	if err != nil {
+		klog.Errorf("NodePodProbe update NodePodProbe(%s) failed:%s", pod.Spec.NodeName, err.Error())
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileEnhancedLivenessProbeMapNodeProbe) retryOnConflictAddNodeProbeConfig(pod *v1.Pod,
+	containersLivenessProbeConfig []livenessprobeUtils.ContainerLivenessProbe) error {
+	nppClone := &appsv1alpha1.NodePodProbe{}
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: pod.Spec.NodeName}, nppClone); err != nil {
+			if errors.IsNotFound(err) {
+				klog.Warningf("No found getting updated npp %s from client", pod.Spec.NodeName)
+				return nil
+			}
+			klog.Errorf("Error getting updated npp %s from client", pod.Spec.NodeName)
+			return err
+		}
+
+		podNewNppCloneSpec := nppClone.Spec.DeepCopy()
+
+		isHit := false
+		isChanged := true
+		for index := range podNewNppCloneSpec.PodProbes {
+			podContainerProbe := &podNewNppCloneSpec.PodProbes[index]
+			if podContainerProbe.Name == pod.Name && podContainerProbe.Namespace == pod.Namespace &&
+				podContainerProbe.UID == fmt.Sprintf("%v", pod.UID) {
+				isHit = true
+				// diff the current pod container probes vs the npp container probes
+				if reflect.DeepEqual(podContainerProbe.Probes, generatePodContainersProbe(pod, containersLivenessProbeConfig)) {
```

Comment:
duplicated calls to generatePodContainersProbe(3 times), just save the result of generatePodContainersProbe in a local variable e.g. newPodContainerProbes, and the isChanged is redundant if newPodContainerProbes is available